backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -120,36 +132,54 @@ public SubjectComment commentAdd(
     }
 
     @Override
-    public SubjectComment commentThumbUp(
+    public SubjectComment commentAddThumbUp(
         @PathVariable Long universityId,
         @PathVariable Long subjectId,
         @PathVariable Long commentId
     ){
-        SubjectComment comment = subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
-        comment.addThumbUp();
-        return subjectCommentRepository.save(comment);
+        SubjectComment result;
+        try{
+            result = subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
+        }catch(NullPointerException|ClassCastException e){
+            throw new SubjectHubException("Specified comment id: "+commentId+", not found for subject id: "+subjectId+
+                ". Unable to add thumb up.");
+        }
+        result.addThumbUp();
```

Comment:
Here is where you'd get the NullPointerException. Because if result is null, it will cause an error by trawing to call .addThumbsUp().

You're try/except block is doing nothing, because an exception isn't thrown from the repository, it either returns an object or it doesn't.

Reply:
oo, fixed that now